src/batchedmul.jl

```diff
@@ -0,0 +1,66 @@
+# batch-wise matrix multiplication
+# wrapper for batched_gemm!
+
+include("./batchedadjtrans.jl")
+
+function batched_mul(A::AbstractArray{T, 3}, B::AbstractArray{T, 3}) where T
+    size(A, 3) == size(B, 3) || throw(DimensionMismatch("batch size mismatch"))
+    batched_mul!(similar(A, (size(A, 1), size(B, 2), size(A, 3))), A, B)
+end
+
+"""
+    batched_mul!(C, A, B) -> C
+batched `mul!`.
+"""
+function batched_mul! end
+
+_unbatch(A) = A
+_unbatch(A::BatchedAdjOrTrans) = A.parent
+
+# bmm
+const _BATCHED_MATRIX_LIST = [
+        (:(AbstractArray{T, 3}), 'N'),
+        (:(BatchedTranspose{T, <:AbstractArray{T, 3}}), 'T'),
+        (:(BatchedAdjoint{T, <:AbstractArray{T, 3}}), 'C')
+]
+
+for (TA, transA) in _BATCHED_MATRIX_LIST, (TB, transB) in _BATCHED_MATRIX_LIST
+    @eval begin
+        function batched_mul!(C::AbstractArray{T, 3}, A::$TA, B::$TB) where T
+            batched_gemm!($transA, $transB, one(T), _unbatch(A), _unbatch(B), zero(T), C)
+            C
+        end
+
+
+    end
+end
+
+function ∇batched_mul(Δ::AbstractArray{T, 3}, A::AbstractArray{T, 3}, B::AbstractArray{T, 3}) where T
+    (batched_mul(Δ, batched_transpose(B)), batched_mul(batched_transpose(A), Δ))
+end
+
+
+function ∇batched_mul(Δ::AbstractArray{T, 3}, A::BatchedTranspose{T, <: AbstractArray{T, 3}}, B::AbstractArray{T, 3}) where T
+    (batched_mul(Δ, batched_transpose(B)), batched_mul(A, Δ))
```

Comment:
Why do you need these extra methods? Don't you need to do `batched_transpose(A)` to unwrap `A`?

Reply:
These extra methods are provided for defining the gradient function, maybe it should be in NNlib.jl? And Yes, the implementation is not correct, I forgot that I'm using `BatchedTranspose`. I need to check that again.

Reply:
I see, but I'm suggesting that only a single gradient definition (the above) is needed, and these ones shouldn't be necessary.

Reply:
Sorry about that, the gradient function is complete wrong. The reason it need several definition is because it's not always `Δ` multiply `A` or `B`. When `A` is transposed, it would `B` multiply  `Δ`. I'll push the correct version later.

Reply:
This still doesn't look correct. You can replace `(delta*b')` with `(b*delta')'` but you're not currently doing the transpose of the output.

In any case, since the original definition `(delta*b')` would calculate the same thing, why have the transposed version at all?

Reply:
Oh, I see. Becuase I didn't define the gradient for `batched_transpose` and `batched_adjoint`.
I'll remove the transposed version. then do I need to define the gradient for `batched_transpose` here? or I just leave it to Flux?